examples/flux_biases_forward.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2025 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+An example to demonstrate simple calibration refinement of flux_biases for
+fast (forward) anneal applied to a full scale simple-cubic spin-glass problem
+with 2-qubit embedding at maximum (extended range) chain strength. 
+"""
+
+
+import argparse
+from typing import Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import dimod
+from hybrid.decomposers import make_origin_embeddings
+from dwave.embedding import embed_bqm
+from dwave.system import DWaveSampler
+
+from dwave.experimental.shimming import shim_flux_biases, qubit_freezeout_alpha_phi
+
+
+def main(
+    solver: Union[None, dict, str],
+    num_iters: int,
+    coupling_strength: float = -1,
+    annealing_time: Union[None, float] = None,
+):
+    """Refine the calibration of a large spin glass.
+
+    See also the calibration refinement tutorial  <https://doi.org/10.3389/fcomp.2023.1238988>
+
+    Args:
+        solver: name of the solver, or dictionary of characteristics.
+        loop_length: length of the loop.
+        num_iters: number of gradient descent steps.
+        coupling_strength: coupling strength on the cubic lattice.
+        annealing_time: annealing_time in microseconds
+    """
+    qpu = DWaveSampler()
```

Comment:
I assume you wanted to use `solver` to initialize the sampler?
```suggestion
    qpu = DWaveSampler(solver=solver)
```